### PR TITLE
Add ability for Vitess to specify client keystore on JDBC connections

### PR DESCRIPTION
**Pull request by nbroyles**

### CODE REVIEW

File: misk-hibernate/src/main/kotlin/misk/jdbc/DataSourceConfig.kt [hunk 1]
```diff
@@ -99,7 +101,34 @@ data class DataSourceConfig(
...
+        var queryParams = ""
+        var useSSL = false
+        /**
+         * Query params for VitessJDBC driver look like default MySQL JDBC driver query params
+         * but are named slightly differently and a smaller subset are supported. See
+         * [io.vitess.jdbc.VitessJDBCUrl] for the complete list
+         */
+        if (!config.trust_certificate_key_store_url.isNullOrBlank()) {
+          require(!config.trust_certificate_key_store_password.isNullOrBlank()) {
+            "must provide a trust_certificate_key_store_password"
```
Comment (wesleyk) [inline misk-hibernate/src/main/kotlin/misk/jdbc/DataSourceConfig.kt]: maybe include "if trust_certificate_key_store_url is set"

File: misk-hibernate/src/main/kotlin/misk/jdbc/DataSourceConfig.kt [hunk 2]
```diff
@@ -99,7 +101,34 @@ data class DataSourceConfig(
...
+          }
+          queryParams += "${if (queryParams.isEmpty()) "?" else "&"}trustStore=${config.trust_certificate_key_store_url}"
```
Comment (wesleyk) [inline misk-hibernate/src/main/kotlin/misk/jdbc/DataSourceConfig.kt]: man github struggles with this syntax highlighting 😬

File: misk-hibernate/src/main/kotlin/misk/jdbc/DataSourceConfig.kt [hunk 3]
```diff
@@ -99,7 +101,34 @@ data class DataSourceConfig(
         "jdbc:hsqldb:mem:${database!!};sql.syntax_mys=true"
       }
       DataSourceType.VITESS -> {
-        "jdbc:vitess://${config.host}:${config.port}/${config.database}"
+        var queryParams = ""
```
Comment (mightyguava) [inline misk-hibernate/src/main/kotlin/misk/jdbc/DataSourceConfig.kt]: I'd just do `queryParams = "?"` here. Should be legal to have a `?` with no params

Comment (nbroyles) [reply misk-hibernate/src/main/kotlin/misk/jdbc/DataSourceConfig.kt]: I'm gonna play it safe for now and leave as-is as I don't have a good way to test that `?` is safe (even though I agree w/ your assessment). Going to work on getting my test app hooked up to a vitess and then I'll try it out.

**Pull request merged**